shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -26,4 +29,25 @@ - (void)touchesBegan:(NSSet*)touches withEvent:(UIEvent*)event {
   }
 }
 
+#if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
+- (void)applicationDidEnterBackground:(UIApplication *)application {
+  // The following keeps the Flutter session alive when the device screen locks
+  // in debug mode. It allows continued use of features like hot reload and 
+  // taking screenshots once the device unlocks again.
+  //
+  // Note the name is not an identifier and multiple instances can exist. 
+  _debugBackgroundTask = [application beginBackgroundTaskWithName:@"Flutter debug task"
+                                                expirationHandler:^{
+      FTL_LOG(WARNING) << "\nThe OS has terminated the Flutter debug connection for being "
+                          "inactive in the background for too long.\n\n"
+                          "There are no errors with your Flutter application.\n\n"
+                          "To reconnect, launch your application again via 'flutter run";
+      }];
+}
+
+- (void)applicationWillEnterForeground:(UIApplication *)application {
+  [application endBackgroundTask: _debugBackgroundTask];
+}
+#endif // FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
```

Comment:
> (we need a formatter)

We use `clang-format` in `buildtools`. There are `clang-format` manifests in the the build root as well as engine repositories. You'd just have to specify the file config. I was just assuming you were using the formatter already.